### PR TITLE
Fixed run-k-points bug by adding change-k-point.

### LINKED ISSUE #770: bug in run-k-points but not run-k-point
There appears to be a slight but reproducible discrepancy between the frequencies generated between (run-k-points) and (run-k-point) for systems with non-zero k.

In the attached file, meep_debug.txt (it's clearly a scheme file but github isn't letting me upload it as .ctl), I'm simulating a photonic crystal slab embedded in a 3D photonic crystal environment comprised of an "extruded" 2D photonic crystal. (The structure is similar to the one shown in Fig. 2a of https://arxiv.org/abs/1901.07126.) The k-point currently specified is that of a bound state in the continuum.

When meep is run using (run-k-points), the frequency of the BIC is found to be 0.36130475184754957.

When meep is instead run using (run-k-point), the frequency of the BIC is found to be 0.35714322640725954.

Although these frequencies are very similar, they are not identical, as one would expect given that (run-k-points) is simply syntactic sugar for (map (lambda (k) (run-k-point)) kList) as is defined in line 1351 of https://github.com/NanoComp/meep/blob/master/scheme/meep.scm.in#L1351. More importantly, when using these frequencies in conjunction with other features available in meep, such as flux regions, this frequency difference is significant, as the Q factor is diverging. Even for non-BIC but high-Q choices of k, the difference in frequencies found between (run-k-points) and (run-k-point) is significant enough to effect results. These other calculations suggest that the correct frequency is found by (run-k-point).

I've done some debugging of this by copying in the definition of (run-k-points), and it appears that the offending line is line #1356 of meep.scm.in, or line 165 of the attached code, which has a call to (init-fields) prior to setting the k-point of the system. Commenting out this one line fixes the discrepancy. So, although I can see that the fields are reset after the k-point is changed in (run-k-point), it would appear that there is some lingering effect from the fields being initialized without the correct k-point.

If you would like, I can put together a quick pull request that deletes this line and moves the (output-epsilon) into the actual run command of (run-k-point), but this clearly isn't the ideal fix. I'd also like to get confirmation that we expect the results of (run-k-point) to be correct, and that you agree there is a slight bug with initializing the fields prior to setting the k-point.

Thanks!

[meep_debug.txt](https://github.com/NanoComp/meep/files/2969209/meep_debug.txt)

### PR DESCRIPTION
**Pull request by acerjan**
Slight update to documentation to remind users to set the k-point prior to adding flux / energy regions.

Resolves #770.

### CODE REVIEW

File: scheme/meep.scm.in [hunk 1]
```diff
@@ -1353,7 +1353,7 @@
   (define all-freqs '())
   (map (lambda (k)
 	 (set! k-index (+ k-index 1))
-	 (if (= k-index 1) (begin (init-fields) (output-epsilon)))
+	 (if (= k-index 1) (begin (change-k-point! k) (init-fields) (output-epsilon)))
```
Comment (stevengj) [inline scheme/meep.scm.in]: `(init-fields)` is no longer required since the fields are initialized by `change-k-point`

**Comment by stevengj** [comment]
I updated it with a few fixes:

* Once you call `change-k-point!` you no longer need to call `init-fields` since the fields are already initialized.
* In Python, we don't call `output_epsilon` by default anymore in other `run` functions, so that whole block can be eliminated.
* In Python, the functions `add_flux` etcetera are now "lazy" (that is, they set up an initialization that is actually performed only when the run begins), so we can remove the language in the manual about only calling them after setting parameters.

**Comment by stevengj** [comment]
Will merge once tests are green again.

**Comment by acerjan** [comment]
When I was testing `change-k-point!` without `init-fields` in scheme, the `output-epsilon` call threw errors complaining that the fields had not been initialized. That's why I left in `init-fields`.

**Comment by stevengj** [comment]
Oh, you are right, `restart-fields` only re-initializes the fields if they have already been initialized.   I updated it to call `init-fields`, but with a guard to ensure that it only re-allocates the fields if needed.

**Pull request merged**